src/app/events/[slug]/utils/filterAndSortScheduleDays.ts

```diff
@@ -0,0 +1,28 @@
+import { UTCDate } from '@date-fns/utc'
+
+import type { Event } from '../../types/eventType'
+
+export function filterAndSortScheduleDays(
+  schedule: NonNullable<Event['schedule']>,
+) {
+  {
+    const daysWithEvents = schedule.days.filter((day) => day.events.length > 0)
+
+    const daysWithEventsSortedByTime = daysWithEvents.map((day) => ({
+      ...day,
+      events: day.events.sort((a, b) => {
+        const startA = new UTCDate(a.start).getTime()
+        const startB = new UTCDate(b.start).getTime()
+        return startA - startB
+      }),
+    }))
+
+    const sortedDays = daysWithEventsSortedByTime.sort((a, b) => {
+      const dateA = new UTCDate(a.date).getTime()
+      const dateB = new UTCDate(b.date).getTime()
+      return dateA - dateB
+    })
+
+    return sortedDays
+  }
+}
```

Comment:
Nit: I've been using `compareAsc` from `'date-fns'` to compare dates asc or desc.

```suggestion
import { UTCDate } from '@date-fns/utc'
import { compareAsc } from 'date-fns'

import type { Event } from '../../types/eventType'

export function filterAndSortScheduleDays(
  schedule: NonNullable<Event['schedule']>,
) {
  {
    const daysWithEvents = schedule.days.filter((day) => day.events.length > 0)

    const daysWithEventsSortedByTime = daysWithEvents.map((day) => ({
      ...day,
      events: day.events.sort((a, b) => {
        const startA = new UTCDate(a.start)
        const startB = new UTCDate(b.start)
        return compareAsc(startA, startB)
      }),
    }))

    const sortedDays = daysWithEventsSortedByTime.sort((a, b) => {
      const dateA = new UTCDate(a.date)
      const dateB = new UTCDate(b.date)
      return compareAsc(dateA, dateB)
    })

    return sortedDays
  }
}
```